scripts/performance/logger.py

```diff
@@ -14,9 +14,27 @@
 
 from .common import get_repo_root_path
 
-__initialized = False
-
-def setup_loggers(verbose: bool):
+class LoggerStateManager:
+    def __init__(self):
+        self.logger_initialized = False
+        self.logger_opentelemetry_imported = False
+
+    def set_initialized(self, value: bool): self.logger_initialized = value
+    def set_opentelemetry_imported(self, value: bool): self.logger_opentelemetry_imported = value
+    def get_initialized(self) -> bool: return self.logger_initialized
+    def get_opentelemetry_imported(self) -> bool: return self.logger_opentelemetry_imported
+
+logger_state_manager = LoggerStateManager()
+
+try:
```

Comment:
When is this code executed? I am unfamiliar with loose code just sitting in a class definition.

Reply:
This code gets executed when the file is imported, though the import will be cached so it is only run once when imported in the same way.